words2num/lang_ES_US.py

```diff
@@ -0,0 +1,267 @@
+ #           ('D', 'T'): f_mul_hundred_and_add,     # 990 (nine ninety)
+ #          ('D', 'M'): f_mul_hundred_and_add,     # 919 (nine nineteen)
+ #           ('T', 'H'): f_mul_hundred,
+            ('T', 'X'): f_mul,     # 90000
+            ('T', 'F'): f_ret,     # 90
+ #           ('M', 'H'): f_mul_hundred,
+            ('M', 'X'): f_mul,     # 19000
+            ('M', 'F'): f_ret,     # 19
+            ('H', 'D'): f_add,     # 909
+            ('H', 'T'): f_add,     # 990
+            ('H', 'M'): f_add,     # 919
+            ('H', 'X'): f_mul,     # 900000
+            ('H', 'F'): f_ret,     # 900
+            ('X', 'D'): f_add,     # 9009
+            ('X', 'T'): f_add,     # 9090
+            ('X', 'M'): f_add,     # 9019
+            ('X', 'H'): f_add,     # 9900
+            ('X', 'F'): f_ret,     # 9000
+            ('Z', 'F'): f_ret,     # 0
+ #           ('A', 'H'): f_mul_hundred,     # 100
+ #           ('A', 'X'): f_mul,      # 1000
+            ('S', 'X'): f_add,      # 1000
+ #           ('A', 'F'): f_ret,      # 1
+        }
+
+    def transition(self, token):
+        value, label = token
+        try:
+            edge_fn = self.edges[(self.state, label)]
+        except KeyError:
+            raise NumberParseException("Invalid number state from "
+                                       "{0} to {1}".format(self.state, label))
+        self.state = label
+        return edge_fn(self, value)
+
+
+def compute_placevalues(tokens):
+    """Compute the placevalues for each token in the list tokens"""
+    pvs = []
+    for tok in tokens:
+        if tok == 'punto':
+            pvs.append(0)
+        else:
+            pvs.append(placevalue(VOCAB[tok][0]))
+    return pvs
+
+def tokenize(text):
+    tokens = re.split(r"[\s,\-]+(?:y)?", text.lower())
+    # Remove empty strings caused by split
+    tokens = [tok for tok in tokens if tok]
+    try:
+        # don't use generator here because we want to raise the exception
+        # here now if the word is not found in vocabulary (easier debug)
+        decimal = False
+        parsed_tokens = []
+        decimal_tokens = []
+        mul_tokens = []
+        pvs = compute_placevalues(tokens)
+        # Loop until all trailing multiplier tokens are removed and added to mul_tokens; Loop conditions:
+        # 1: The last token in the list must have the highest placevalue of any token
+        # 2: The list of tokens must be longer than one (to prevent extracting all tokens as mul_tokens)
+        # 3: The maximum placevalue must be greater than 1 (This limits our mul_tokens to "hundred" or greater)
+        while max(pvs) == pvs[-1] and len(pvs) > 1 and max(pvs) > 1:
+            mul_tokens.insert(0, VOCAB[tokens.pop()])
+            pvs.pop()
+
+        for token in tokens:
+            if token == 'punto':
+                if decimal:
+                    raise ValueError("Invalid decimal word "
+                                        "'{0}'".format(token))
```

Comment:
f-string here too? why is the line broken? is it over the char limit?

Reply:
Changed line as follows: raise ValueError("Invalid decimal word '{token}'")